### PR TITLE
[gui2] Merge DesktopBrowserLauncher into Utilities

### PR DESCRIPTION
**Pull request by grimreaper**
The utilities code appears to be a pass-through into the desktop code, so
just merge them all together and make the logic slightly less complex.

### CODE REVIEW

File: code/src/java/pcgen/gui2/converter/panel/RunConvertPanel.java [hunk 1]
```diff
@@ -125,18 +126,13 @@ public boolean performAnalysis(final CDOMObject pc)
...
-					}
-				}
-			}
+			campaign.getSafeListFor(ListKey.FILE_PCC)
+			        .stream()
+			        .map(CampaignSourceEntry::getURI)
+			        .filter(PCGFile::isPCGenCampaignFile)
+			        .map(uri -> Globals.getCampaignByURI(uri, false))
+			        .filter(Objects::nonNull)
+			        .forEach(c -> totalCampaigns.add(c));
```
Comment (karianna) [inline code/src/java/pcgen/gui2/converter/panel/RunConvertPanel.java]: what's a `c`? :-)

File: code/src/java/pcgen/gui2/converter/panel/RunConvertPanel.java [hunk 2]
```diff
@@ -540,17 +536,9 @@ public void run()
...
-		aSelectedCampaignsList.sort(new Comparator<Campaign>()
-		{
-			@Override
-			public int compare(Campaign c1, Campaign c2)
-			{
-				return c1.getSafe(IntegerKey.CAMPAIGN_RANK) - c2.getSafe(IntegerKey.CAMPAIGN_RANK);
-			}
-
-		});
+		aSelectedCampaignsList.sort(Comparator.comparingInt(c -> c.getSafe(IntegerKey.CAMPAIGN_RANK)));
```
Comment (karianna) [inline code/src/java/pcgen/gui2/converter/panel/RunConvertPanel.java]: what's a c

File: code/src/java/pcgen/gui2/dialog/AboutDialog.java [hunk 3]
```diff
@@ -225,44 +227,30 @@ private JPanel buildCreditsPanel()
 
 		// Web site button
 		wwwSite.setText(PCGenPropBundle.getWWWHome());
-		wwwSite.addActionListener(new ActionListener()
-		{
-
-			@Override
-			public void actionPerformed(ActionEvent e)
+		wwwSite.addActionListener(e -> {
```
Comment (karianna) [inline code/src/java/pcgen/gui2/dialog/AboutDialog.java]: e is an event

File: code/src/java/pcgen/gui2/facade/SpellSupportFacadeImpl.java [hunk 4]
```diff
@@ -93,6 +91,8 @@
 import pcgen.util.enumeration.Tab;
 import pcgen.util.enumeration.View;
 
+import org.apache.commons.lang3.StringUtils;
```
Comment (karianna) [inline code/src/java/pcgen/gui2/facade/SpellSupportFacadeImpl.java]: why this import?

File: code/src/java/pcgen/gui2/prefs/LocationPanel.java [hunk 5]
```diff
@@ -17,6 +17,7 @@
  */
 package pcgen.gui2.prefs;
 
+import java.awt.Desktop;
```
Comment (karianna) [inline code/src/java/pcgen/gui2/prefs/LocationPanel.java]: why this import?

File: code/src/java/pcgen/gui2/dialog/AboutDialog.java [hunk 6]
```diff
@@ -225,44 +227,30 @@ private JPanel buildCreditsPanel()
...
-				}
-				catch (IOException ioe)
-				{
-					Logging.errorPrint(LanguageBundle.getString("in_abt_browser_err"), ioe); //$NON-NLS-1$
-				}
+				DesktopBrowserLauncher.viewInBrowser(new URL(mailingList.getText()));
+			}
+			catch (IOException ioe)
+			{
+				Logging.errorPrint(LanguageBundle.getString("in_abt_browser_err"), ioe); //$NON-NLS-1$
```
Comment (karianna) [inline code/src/java/pcgen/gui2/dialog/AboutDialog.java]: Should be `in_err_browser_err`

**Comment by grimreaper** [comment]
resolved.

**Comment by karianna** [comment]
@grimreaper Travis failures again.  Can you run the PMD target before submitting PRs going forwards, will speed things up.

**Comment by grimreaper** [comment]
Don't worry about commenting on travis-failed PRs: I usually know about them already.  Just treat them as non-PRs.

In terms of running pmd in advance: I try to, though sometimes the process I have of linear commits following cherry picking the changes misses stuff.

**Pull request merged**